extensions/ql-vscode/src/remote-queries/variant-analysis-processor.ts

```diff
@@ -0,0 +1,157 @@
+  response: ApiVariantAnalysis
+): VariantAnalysis {
+
+  let scannedRepos: VariantAnalysisScannedRepository[] = [];
+  let skippedRepos: VariantAnalysisSkippedRepositories = {};
+
+  if (response.scanned_repositories) {
+    scannedRepos = processScannedRepositories(response.scanned_repositories as ApiVariantAnalysisScannedRepository[]);
+  }
+
+  if (response.skipped_repositories) {
+    skippedRepos = processSkippedRepositories(response.skipped_repositories as ApiVariantAnalysisSkippedRepositories);
+  }
+
+  const variantAnalysis: VariantAnalysis = {
+    id: response.id,
+    controllerRepoId: response.controller_repo.id,
+    query: {
+      name: submission.query.name,
+      filePath: submission.query.filePath,
+      language: submission.query.language
+    },
+    databases: submission.databases,
+    status: processApiStatus(response.status),
+    actionsWorkflowRunId: response.actions_workflow_run_id,
+    scannedRepos: scannedRepos,
+    skippedRepos: skippedRepos
+  };
+
+  if (response.failure_reason) {
+    variantAnalysis.failureReason = response.failure_reason as VariantAnalysisFailureReason;
+  }
+
+  return variantAnalysis;
+}
+
+function processScannedRepositories(
+  scannedRepos: ApiVariantAnalysisScannedRepository[]
+): VariantAnalysisScannedRepository[] {
+
+  const result: VariantAnalysisScannedRepository[] = [];
+
+  scannedRepos.forEach(function(scannedRepo) {
+    const parsedRepo: VariantAnalysisScannedRepository = {
+      repository: {
+        id: scannedRepo.repository.id,
+        fullName: scannedRepo.repository.full_name,
+        private: scannedRepo.repository.private,
+      },
+      analysisStatus: processApiRepoStatus(scannedRepo.analysis_status),
+      resultCount: scannedRepo.result_count,
+      artifactSizeInBytes: scannedRepo.artifact_size_in_bytes,
+      failureMessage: scannedRepo.failure_message
+    };
+
+    result.push(parsedRepo);
+  });
+
+  return result;
+}
+
+function processSkippedRepositories(
+  skippedRepos: ApiVariantAnalysisSkippedRepositories
+): VariantAnalysisSkippedRepositories {
+
+  return {
+    accessMismatchRepos: processRepoGroup(skippedRepos.access_mismatch_repos),
+    notFoundRepos: processNotFoundRepoGroup(skippedRepos.not_found_repo_nwos),
+    noCodeqlDbRepos: processRepoGroup(skippedRepos.no_codeql_db_repos),
+    overLimitRepos: processRepoGroup(skippedRepos.over_limit_repos)
+  };
+}
+
+function processRepoGroup(repoGroup: ApiVariantAnalysisSkippedRepositoryGroup): VariantAnalysisSkippedRepositoryGroup {
+  const repos = repoGroup.repositories.map(repo => {
+    return {
+      id: repo.id,
+      fullName: repo.full_name
+    };
+  });
+
+  return {
+    repositoryCount: repoGroup.repository_count,
+    repositories: repos
+  };
+}
+
+function processNotFoundRepoGroup(repoGroup: ApiVariantAnalysisNotFoundRepositoryGroup): VariantAnalysisSkippedRepositoryGroup {
+  const repo_nwos = repoGroup.repository_nwos.map(nwo => {
```

Comment:
Thanks, have changed now. 

There's still the `VariantAnalysisSkippedRepositories` which has a key called `not_found_repo_nwos`. I'm going to leave it as is since it doesn't look like we've changed the API for this key.

Reply:
Thanks!

I don't think the API is still using `not_found_repo_nwos` anywhere. From the REST API documentation it seems like that field should be called `not_found_repos`.